Final2-Another Candy Crush Clone/edu/kit/informatik/matchthree/tests/MyMaximumDeltaMatcherTest.java

```diff
@@ -0,0 +1,167 @@
+                Position.at(1, 0),
+                Position.at(2, 1),
+                Position.at(1, 2)
+        ));
+        expectedMatches.add(match2);
+        assertSetOfSetsEquals(expectedMatches, actualMatches);
+    }
+    
+    @Test
+    public void myTest3() {
+        final Board board = new MatchThreeBoard(Token.set("ABC"), "B B;CAB;ABA");
+        Delta delta1 = new Delta(1,1);
+        Delta delta2 = new Delta(1,-1);
+        Set<Delta> deltas = new HashSet<Delta>();
+        deltas.add(delta1);
+        deltas.add(delta2);
+        Set<Position> initial = new HashSet<Position>();
+        initial.add(new Position(0,2));
+        initial.add(new Position(1,0));
+        final Set<Set<Position>> actualMatches = new MaximumDeltaMatcher(deltas).matchAll(board, initial);
+
+        Set<Set<Position>> expectedMatches = new HashSet<>();
+        final Set<Position> match = new HashSet<>(Arrays.asList(
+                Position.at(0, 2),
+                Position.at(1, 1),
+                Position.at(2, 2)
+        ));
+        expectedMatches.add(match);
+        Set<Position> match2 = new HashSet<Position>();
+        expectedMatches.add(match2);
+        assertSetOfSetsEquals(expectedMatches, actualMatches);
+    }
+    
+    @Test
+    public void myTest4() {
+        final Board board = new MatchThreeBoard(Token.set("ABC"), "B B;CAB;ABA");
+        Delta delta1 = new Delta(1,1);
+        Delta delta2 = new Delta(1,-1);
+        Set<Delta> deltas = new HashSet<Delta>();
+        deltas.add(delta1);
+        deltas.add(delta2);
+        Set<Position> initial = new HashSet<Position>();
+        initial.add(new Position(0,2));
+        initial.add(new Position(1,0));
+        initial.add(new Position(1,2));
+        final Set<Set<Position>> actualMatches = new MaximumDeltaMatcher(deltas).matchAll(board, initial);
+
+        Set<Set<Position>> expectedMatches = new HashSet<>();
+        final Set<Position> match = new HashSet<>(Arrays.asList(
+                Position.at(0, 2),
+                Position.at(1, 1),
+                Position.at(2, 2)
+        ));
+        expectedMatches.add(match);
+        Set<Position> match2 = new HashSet<Position>();
+        expectedMatches.add(match2);
+        final Set<Position> match3 = new HashSet<>(Arrays.asList(
+                Position.at(1, 2),
+                Position.at(2, 1)
+        ));
+        expectedMatches.add(match3);
+        assertSetOfSetsEquals(expectedMatches, actualMatches);
+    }
+    
+    public void myTest5() {
+        final Board board = new MatchThreeBoard(Token.set("ABC"), "B B; AB;ABA");
+        Delta delta1 = new Delta(1,1);
+        Delta delta2 = new Delta(1,-1);
+        Set<Delta> deltas = new HashSet<Delta>();
+        deltas.add(delta1);
+        deltas.add(delta2);
+        Set<Position> initial = new HashSet<Position>();
+        initial.add(new Position(0,2));
+        initial.add(new Position(1,0));
```

Comment:
Yes, in Java terms two tokens X and X are different _objects_, but they represent the same _token_ as of Token#equals().